Automattic/wp-calypso
# Add Individual Subscriber Stats

![CleanShot 2023-08-15 at 11 17 07@2x](https://github.com/Automattic/wp-calypso/assets/528287/f3497c26-1de0-4ec0-af84-524e70971d0b)


## Proposed Changes

This PR adds the boxes for individual subscriber stats to the subscriber detail page.

- It adds an `individual-subscriber-stats` flag, that's only enabled in development
- It adds a `<SubscriberStatsCard />` component, which is closely related to the HighlightCard component, but adds support for non-numerical values, loading state & tooltip support.
- It adds the `useSubscriberStatsQuery` react-query
- It adds the `<SubscriberStats />` component, which calls the query and renders the 3 boxes.
- It adds the needed icons

### Housekeeping
- This PR fixes how props are passed to the pages inside `subscribers/controller.js`. Previously it used `context.path`, which also includes the query string. By doing `const userId = path.split( '/' ).pop();` on `/subscribers/123456?page=1`, this would end up with a userId with value `123456?page=1`. I updated this to use `context.params`, which seems like the way to do this proper.
- This PR creates an export for { useSubscriberDetailsQuery } inside `queries/index.ts`, as this is how we've exported/imported queries before. It also updates the import inside `subscriber-details-page.tsx`
- This PR creates an export for { useSubscriptionPlans } inside `hooks/index.ts`, for the same reason. It also updates the imports.

## Testing Instructions

- First, apply this PR to your sandbox: D118596-code
- Sandbox `public-api.wordpress.com` 
- Apply this PR & `yarn start`
- Visit `Users > Subscribers`
- Click an external user, verify that the subscriber stats load
- Click on a WPCOM user, verify that the subscriber stats load

## Pre-merge Checklist
